tee-worker/ts-tests/integration-tests/data-providers.test.ts

```diff
@@ -0,0 +1,111 @@
+import { assertIsInSidechainBlock, assertVc } from './common/utils/assertion';
+import {
+    getSidechainNonce,
+    getTeeShieldingKey,
+    sendRequestFromTrustedCall,
+    createSignedTrustedCallRequestVc,
+} from './common/di-utils'; // @fixme move to a better place
+import type { IntegrationTestContext } from './common/common-types';
+import { CorePrimitivesIdentity, RequestVCResult } from 'parachain-api';
+import { aesKey } from './common/call';
+import { $ as zx } from 'zx';
+import { credentialDefinitionMap } from './common/credential-definitions';
+import { subscribeToEventsWithExtHash } from './common/transactions';
+describe('Test Vc (direct invocation)', function () {
+    let context: IntegrationTestContext = undefined as any;
+    let teeShieldingKey: KeyObject = undefined as any;
+    let aliceSubstrateIdentity: CorePrimitivesIdentity = undefined as any;
+    const client = process.env.BINARY_DIR + '/litentry-cli';
+    const aliceAddressFormat = '0xd43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d';
+    const reqExtHash = '0x0000000000000000000000000000000000000000000000000000000000000000';
+    let argvId = '';
+    this.timeout(6000000);
+    before(async () => {
+        context = await initIntegrationTestContext(process.env.WORKER_ENDPOINT!, process.env.NODE_ENDPOINT!);
+        teeShieldingKey = await getTeeShieldingKey(context);
+        aliceSubstrateIdentity = await buildIdentityFromKeypair(
+            new PolkadotSigner(context.substrateWallet.alice),
+            context
+        );
+    });
+
+    // usage example:
+    // pnpm run test-data-providers:local --id=vip3-membership-card-gold for single test
+    // pnpm run test-data-providers:local for all tests
+    const argv = process.argv.indexOf('--id');
+    argvId = process.argv[argv + 1];
+    // eslint-disable-next-line no-prototype-builtins
+    if (argvId && credentialDefinitionMap.hasOwnProperty(argvId)) {
+        step(`linking identity-${credentialDefinitionMap[argvId].mockAddress} via cli`, async function () {
+            const eventsPromise = subscribeToEventsWithExtHash(reqExtHash, context);
+            try {
+                await zx`${client} trusted -d link-identity did:litentry:substrate:${aliceAddressFormat}\
+            did:${credentialDefinitionMap[argvId].mockDid}\
+            ${credentialDefinitionMap[argvId].mockWeb3Network}`;
+            } catch (error: any) {
+                console.log(`Exit code: ${error.exitCode}`);
+                console.log(`Error: ${error.stderr}`);
+                throw error;
+            }
+
+            const events = (await eventsPromise).map(({ event }) => event);
+            assert.equal(events.length, 1);
+        });
+    } else {
+        Object.keys(credentialDefinitionMap).forEach((id) => {
+            step(`linking identity-${credentialDefinitionMap[id].mockAddress} via cli`, async function () {
```

Comment:
Can you extract and reuse this function ? I think it's exactly the same to one from line 43.